engine/src/mbliphonedc.mm

```diff
@@ -1203,6 +1243,12 @@ static void MCIPhoneDoDidBecomeActive(void *)
 	t_init_success = X_init(1, args, env);
 	
 	[t_pool release];
+    
+    // Release the env array and its elements - X_init only pass env to X_open,
+    //  which only copies the relevant env vars from it: safe to release here.
+    for (uindex_t j = 0; j < envc; ++j)
+        MCMemoryDelete(env[j]);
```

Comment:
In the (unlikely) event that MCMemoryNewArray(envc +1, env) fails on line 1202, accessing env[j] will result in a SEGFAULT.